internal/controller/managedcluster_controller.go

```diff
@@ -303,24 +308,129 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
+			continue
+		}
+
+		tmpl := &hmc.ServiceTemplate{}
+		tmplRef := types.NamespacedName{Name: svc.Template, Namespace: mc.Namespace}
+		if err := r.Get(ctx, tmplRef, tmpl); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get Template (%s)", tmplRef.String())
+		}
+
+		source, err := r.getServiceTemplateSource(ctx, tmpl)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not get repository url: %w", err)
+		}
+
+		opts = append(opts, sveltos.HelmChartOpts{
+			RepositoryURL:  source.Spec.URL,
+			RepositoryName: tmpl.Spec.Helm.ChartName,
+			ChartName:      tmpl.Spec.Helm.ChartName,
+			ChartVersion:   tmpl.Spec.Helm.ChartVersion,
+			ReleaseName:    svc.Name,
+			Values:         svc.Values,
+			ReleaseNamespace: func() string {
+				if svc.Namespace != "" {
+					return svc.Namespace
+				}
+				return svc.Name
+			}(),
+			// The reason it is passed to PlainHTTP instead of InsecureSkipTLSVerify is because
+			// the source.Spec.Insecure field is meant to be used for connecting to repositories
+			// over plain HTTP, which is different than what InsecureSkipTLSVerify is meant for.
+			// See: https://github.com/fluxcd/source-controller/pull/1288
+			PlainHTTP: source.Spec.Insecure,
+		})
+	}
+
+	if _, _, err := sveltos.ReconcileProfile(ctx, r.Client, l, mc.Namespace, mc.Name,
+		map[string]string{
+			hmc.FluxHelmChartNamespaceKey: mc.Namespace,
+			hmc.FluxHelmChartNameKey:      mc.Name,
+		},
+		sveltos.ReconcileProfileOpts{
+			OwnerReference: &metav1.OwnerReference{
+				APIVersion: hmc.GroupVersion.String(),
+				Kind:       hmc.ManagedClusterKind,
+				Name:       mc.Name,
+				UID:        mc.UID,
+			},
+			HelmChartOpts: opts,
+		}); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to reconcile Profile: %w", err)
+	}
+
+	// We don't technically need to requeue here, but doing so because golint fails with:
+	// `(*ManagedClusterReconciler).updateServices` - result `res` is always `nil` (unparam)
+	//
+	// This will be automatically resolved once setting status is implemented (https://github.com/Mirantis/hmc/issues/361),
+	// as it is likely that some execution path in the function will have to return with a requeue to fetch latest status.
+	return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
+}
+
+// getServiceTemplateSource returns the source (HelmRepository) used by the ServiceTemplate.
+// It is fetched by querying for ServiceTemplate -> HelmChart -> HelmRepository.
+func (r *ManagedClusterReconciler) getServiceTemplateSource(ctx context.Context, tmpl *hmc.ServiceTemplate) (*sourcev1.HelmRepository, error) {
+	tmplRef := types.NamespacedName{Namespace: tmpl.Namespace, Name: tmpl.Name}
+
+	if tmpl.Status.ChartRef == nil {
+		return nil, fmt.Errorf("status for ServiceTemplate (%s) has not been updated yet", tmplRef.String())
+	}
+
+	hc := &sourcev1.HelmChart{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: tmpl.Status.ChartRef.Namespace,
+		Name:      tmpl.Spec.Helm.ChartName,
```

Comment:
Can we also get the chart name from the `Status.ChartRef`? Template spec API supports configuring either `chartName` or `chartRef`: https://github.com/Mirantis/hmc/blob/main/api/v1alpha1/templates_common.go#L43
So, `chartName` may be unset.